### ISSUE TITLE
How to suppress response schema for response code = 204 No Content?

### ISSUE THREAD

**Issue by dthorpe** [domaindrivendev/Swashbuckle#520]
I have an async Web API controller method that uses ````Task<IHttpActionResult>```` as the result type for convenience in generating detailed error responses.

Swashbuckle considers this a non-void function return type, and emits a response schema referencing "Object" in the output JSON.

How do I suppress that schema output, so that swagger-UI doc for this function will look the same as what is shown for a void function?

Example:
````c#
/// <summary>
/// description text
/// </summary>
/// <response code="204">No Content</response>
/// <response code="400">Bad Request - Invalid or missing parameter data</response>
[HttpDelete]
public async Task<IHttpActionResult> DeleteContract(Guid walletId, Guid contractId)
{ }
````

This currently outputs:
````javascript
responses: {
204: {
description: "No Content",
schema: {
$ref: "#/definitions/Object"
}
},
400: {
description: "Bad Request - Invalid or missing parameter data"
},
}
````

But I want it to look like this:
````javascript
responses: {
204: {
description: "No Content",
},
400: {
description: "Bad Request - Invalid or missing parameter data"
},
}
````

As a side note, shouldn't Swashbuckle do this return type suppression automatically for response code = 204?

**Comment by domaindrivendev** [comment]
Good catch - I think it's reasonable to say SB should do this automatically. Fancy submitting a PR?

**Comment by dthorpe** [comment]
Ya, I think I figure out a code fix this weekend.  Which class would be the best place to make the change?

**Comment by domaindrivendev** [comment]
The update needs to happen somewhere around the following line ...

https://github.com/domaindrivendev/Swashbuckle/blob/master/Swashbuckle.Core/Swagger/XmlComments/ApplyXmlActionComments.cs#L96

**Comment by davidkeaveny** [comment]
I worked around this by writing my own ResponseType attributes, starting with a base class:

```
[AttributeUsage(AttributeTargets.Method, AllowMultiple = true, Inherited = false)]
public class ResponseTypeAttribute : Attribute
{
    public ResponseTypeAttribute(HttpStatusCode statusCode)
        : this(statusCode, typeof(void))
    {
    }

    public ResponseTypeAttribute(HttpStatusCode statusCode, Type response)
    {
        StatusCode = statusCode;
        Response = response;
    }

    public HttpStatusCode StatusCode { get; set; }
    public Type Response { get; set; }
}
```

I could either use the class as is, or sub-class it for more conciseness and consistency:

```
[AttributeUsage(AttributeTargets.Method, AllowMultiple = false, Inherited = false)]
public class ResponseTypeNoContentAttribute : Attribute
{
    public ResponseTypeNoContentAttribute()
        : base(HttpStatusCode.NoContent)
}
```

I then wrote an IOperationFilter that reads the attributes and updates the definition accordingly, looking at the response type. 

I find this far more powerful than using the XMLDOC comments - for starters (and I haven't shown it here) it allows me to use resources for internationlised descriptions, either by convention or explicitly provided. It also gives you compile-time checking for those times when someone renames a response class. I can also automatically add common response types, such as 500 or 503 to all responses.

Finally, for good measure, I later added the ability to provide example response data, in a similar manner to [@mattfrear](https://mattfrear.com/2015/04/21/generating-swagger-example-responses-with-swashbuckle/)

So the final result is something like this:

[ResponseTypeNoContent]
[ResponseTypeBadRequest]
public IHttpActionResult Put(long id, UpdateCommand command)
{
    // Do stuff here
    return new NoContentResult();
}

And nary an XML comment to be seen, at least as far as the response types are concerned.
